### PR TITLE
Add table aws_redshift_event_subscription. Closes #219

### LINKED ISSUE #219: Add table aws_redshift_event_subscription
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.
https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.38.0/service/redshift/redshiftiface

### PR DESCRIPTION
**Pull request by anisadas**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_redshift_event_subscription []

PRETEST: tests/aws_redshift_event_subscription

TEST: tests/aws_redshift_event_subscription
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_sns_topic.test_resource: Creating...
aws_sns_topic.test_resource: Creation complete after 4s [id=arn:aws:sns:us-east-1:986325076436:redshift-events]
aws_redshift_event_subscription.named_test_resource: Creating...
aws_redshift_event_subscription.named_test_resource: Creation complete after 2s [id=turbottest46694]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:redshift:us-east-1:986325076436:eventsubscription:turbottest46694
resource_name = turbottest46694

Running SQL query: test-get-query.sql
[
  {
    "account_id": "986325076436",
    "cust_subscription_id": "turbottest46694",
    "partition": "aws",
    "region": "us-east-1",
    "tags_src": [
      {
        "Key": "Name",
        "Value": "turbottest46694"
      }
    ],
    "title": "turbottest46694"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "account_id": "986325076436",
    "cust_subscription_id": "turbottest46694",
    "partition": "aws",
    "region": "us-east-1",
    "title": "turbottest46694"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:redshift:us-east-1:986325076436:eventsubscription:turbottest46694"
    ],
    "tags": {
      "Name": "turbottest46694"
    },
    "title": "turbottest46694"
  }
]
✔ PASSED

POSTTEST: tests/aws_redshift_event_subscription

TEARDOWN: tests/aws_redshift_event_subscription

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
select
  cust_subscription_id,
  customer_aws_id,
  status,
  sns_topic_arn,
  subscription_creation_time
from
  aws_redshift_event_subscription;

+----------------------+-----------------+--------+--------------------------------------------------+----------------------------+
| cust_subscription_id | customer_aws_id | status | sns_topic_arn                                    | subscription_creation_time |
+----------------------+-----------------+--------+--------------------------------------------------+----------------------------+
| turbottest1234       | 986325076436    | active | arn:aws:sns:us-east-1:986325076436:dynamodb      | 2021-03-19 11:13:28        |
| test-event-query     | 986325076436    | active | arn:aws:sns:us-east-2:986325076436:test-redshift | 2021-03-19 10:16:42        |
| event-test11         | 986325076436    | active | arn:aws:sns:us-east-2:986325076436:test-redshift | 2021-03-19 06:28:16        |
| test-steampipi       | 986325076436    | active | arn:aws:sns:us-east-1:986325076436:dynamodb      | 2021-03-19 10:07:13        |
+----------------------+-----------------+--------+--------------------------------------------------+----------------------------+


select
  cust_subscription_id,
  customer_aws_id,
  status,
  enabled,
  sns_topic_arn,
  subscription_creation_time
from
  aws_redshift_event_subscription
where
  enabled is false;

+----------------------+-----------------+--------+---------+--------------------------------------------------+----------------------------+
| cust_subscription_id | customer_aws_id | status | enabled | sns_topic_arn                                    | subscription_creation_time |
+----------------------+-----------------+--------+---------+--------------------------------------------------+----------------------------+
| test-event-query     | 986325076436    | active | false   | arn:aws:sns:us-east-2:986325076436:test-redshift | 2021-03-19 10:16:42        |
| turbottest1234       | 986325076436    | active | false   | arn:aws:sns:us-east-1:986325076436:dynamodb      | 2021-03-19 11:13:28        |
+----------------------+-----------------+--------+---------+--------------------------------------------------+----------------------------+

select
  cust_subscription_id,
  severity,
  source_type,
  event_categories_list,
  source_ids_list
from
  aws_redshift_event_subscription;

+----------------------+----------+-------------------------+--------------------------------------------------------+---------------------+
| cust_subscription_id | severity | source_type             | event_categories_list                                  | source_ids_list     |
+----------------------+----------+-------------------------+--------------------------------------------------------+---------------------+
| event-test11         | INFO     | <null>                  | <null>                                                 | <null>              |
| test-event-query     | INFO     | cluster-parameter-group | ["management","monitoring"]                            | <null>              |
| turbottest1234       | ERROR    | cluster-snapshot        | ["management","pending","monitoring"]                  | ["turbottest80541"] |
| test-steampipi       | ERROR    | cluster                 | ["security","management","configuration","monitoring"] | <null>              |
+----------------------+----------+-------------------------+--------------------------------------------------------+---------------------+


select
  e.cust_subscription_id,
  e.status,
  s.kms_master_key_id,
  s.topic_arn as arn
from
  aws_redshift_event_subscription as e
  join aws_sns_topic as s on s.topic_arn = e.sns_topic_arn
where
  s.kms_master_key_id is null;

+----------------------+--------+-------------------+------------------------------------------------------+
| cust_subscription_id | status | kms_master_key_id | arn                                                  |
+----------------------+--------+-------------------+------------------------------------------------------+
| test-steam           | active | <null>            | arn:aws:sns:us-east-2:986325076436:dynamodb          |
| test-anisa           | active | <null>            | arn:aws:sns:us-east-2:986325076436:turbot-topic-test |
+----------------------+--------+-------------------+------------------------------------------------------+
```
</details>

### CODE REVIEW

File: aws-test/tests/aws_redshift_event_subscription/test-get-query.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+select account_id , cust_subscription_id, partition , region , tags_src , title
```
Comment (Paulami30) [inline aws-test/tests/aws_redshift_event_subscription/test-get-query.sql:1]:
```suggestion
select account_id, cust_subscription_id, partition, region, tags_src, title
```
```suggestion
select account_id , cust_subscription_id, partition , region , tags_src , title
```

Comment (Paulami30) [reply aws-test/tests/aws_redshift_event_subscription/test-get-query.sql:1]: please remove these unwanted space in the sql queries

File: docs/tables/aws_redshift_event_subscription.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  sns_topic_arn,
+  subscription_creation_time
+from
+  aws_redshift_event_subscription
+where
+  enabled is false;
+```
+
+
+### Details of the source associated with event subscription
```
Comment (Paulami30) [inline docs/tables/aws_redshift_event_subscription.md:38]:
```suggestion
### Get details of the source associated with event subscriptions
```
```suggestion
### Details of the source associated with event subscription
```

File: aws/table_aws_redshift_event_subscription.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+	if strings.HasPrefix(*parameterData.CustSubscriptionId, ":") {
+		aka = aka + *parameterData.CustSubscriptionId
+	} else {
+		aka = aka + ":" + *parameterData.CustSubscriptionId
+	}
+
+	return []string{aka}, nil
+}
+
```
Comment (Paulami30) [inline aws/table_aws_redshift_event_subscription.go:197]:
```suggestion

//// TRANSFORM FUNCTIONS

```

File: docs/tables/aws_redshift_event_subscription.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# Table: aws_redshift_event_subscription
+
+Amazon Redshift Event Subscription notifies when an event occurs for a cluster,snapshot,security group, or parameter group.
```
Comment (Paulami30) [inline docs/tables/aws_redshift_event_subscription.md:3]:
```suggestion
Amazon Redshift Event Subscription notifies when an event occurs for a cluster, snapshot, security group or parameter group.
```

File: docs/tables/aws_redshift_event_subscription.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  status,
+  enabled,
+  sns_topic_arn,
+  subscription_creation_time
+from
+  aws_redshift_event_subscription
+where
+  enabled is false;
+```
+
```
Comment (Paulami30) [inline docs/tables/aws_redshift_event_subscription.md:36]: Add a query to check whether the sns topic it is using is encrypted or not

**Review by Paulami30 — COMMENTED** [review]
check the comments and resolve merge conflict

File: docs/tables/aws_redshift_event_subscription.md (hunk 5 above)
Comment (Paulami30) [reply docs/tables/aws_redshift_event_subscription.md:36]: can you check this suggestion also?

**Comment by Paulami30** [comment]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by Paulami30 — APPROVED** [review]

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**